src/node_url.cc

```diff
@@ -162,23 +170,17 @@ namespace url {
               } else {
                 value = value * 10 + number;
               }
-              if (value > 255)
-                goto end;
               pointer++;
               ch = pointer < end ? pointer[0] : kEOL;
+              if (value > 255)
+                goto end;
             }
-            if (dots < 3 && ch != '.')
-              goto end;
             *piece_pointer = *piece_pointer * 0x100 + value;
-            if (dots & 0x1)
+            numbers_seen++;
+            if (numbers_seen == 2 || numbers_seen == 4)
               piece_pointer++;
-            if (ch != kEOL) {
-              pointer++;
-              ch = pointer < end ? pointer[0] : kEOL;
-            }
-            if (dots == 3 && ch != kEOL)
+            if (ch == kEOL && numbers_seen != 4)
```

Comment:
I think it could happen if the `numbers_seen` is increased in the loop in the loop that the top loop can't detect: https://github.com/watilde/node/blob/521926ae2f502759c5fc752c82a2661a3dbf419e/src/node_url.cc#L179